### PR TITLE
Added cover for handling memory.

### PR DESCRIPTION
**Pull request by simonmika**
This PR is for illustrating a concept.

### CODE REVIEW

File: source/base/Memory.ooc [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		this init(null, 0)
+	}
+	init: func@ ~fromSize (size: Int) {
+		this init(gc_malloc(size), size)
+	}
+	init: func@ ~fromData (=_pointer, =_size)
+	free: func@ -> Bool {
+		result: Bool
+		if (result = this _pointer != null)
+			gc_free(this _pointer)
```
Comment (davidhesselbom) [inline source/base/Memory.ooc]: And if we want to get rid of `gc_free`?

Comment (simonmika) [inline source/base/Memory.ooc]: That is just a name and as long as we don't change the compiler we will need to define it.

Comment (davidhesselbom) [inline source/base/Memory.ooc]: What do you mean we need to define it? Are you really sure about that?

Comment (marcusnaslund) [inline source/base/Memory.ooc]: Not true. We need some kind of free function of course, but we can name it however we like.

**Comment by marcusnaslund** [comment]
I will close this for now - and revisit/reopen when we're not in a PR rush and when we've decided how we actually want to solve this (either with a `Memory` cover or by extending `Pointer`, which I think would prefer)

**Comment by simonmika** [comment]
@sebastianbaginski You might be interested in this.

**Pull request closed without merging**